bundles/NewsletterBundle/public/js/startup.js

```diff
@@ -0,0 +1,119 @@
+pimcore.registerNS("pimcore.bundle.newsletter.startup");
+
+pimcore.bundle.newsletter.startup = Class.create({
+    type: 'newsletter',
+
+    initialize: function () {
+        document.addEventListener(pimcore.events.prepareDocumentTreeContextMenu, this.onPrepareDocumentTreeContextMenu.bind(this));
+    },
+
+    onPrepareDocumentTreeContextMenu: function (e) {
+        var document = e.detail.document;
+        var menu = e.detail.menu;
+        var tree = e.detail.tree;
+        var me = this;
+
+        if (tree.tree.getSelectionModel().getSelected().length > 1) {
+            return;
+        }
+
+        var documentMenu = {
+            newsletter: [],
+        };
+
+        if (tree.tree.getSelectionModel().getSelected().length > 1) {
+            return;
+        }
+
+        var childSupportedDocument = (document.data.type == "page" || document.data.type == "folder"
+            || document.data.type == "link" || document.data.type == "hardlink"
+            || document.data.type == "printcontainer" || document.data.type == "headlessdocument");
+
+
+        // do not add the newsletter under print containers
+        if(childSupportedDocument && document.data.permissions.create && !pimcore.helpers.documentTypeHasSpecificRole(document.data.type, "only_printable_childrens")) {
+            documentMenu = this.populatePredefinedDocumentTypes(documentMenu, tree, document);
+            // empty newsletter
+            documentMenu.newsletter.push({
```

Comment:
need to add a perspective check
https://github.com/pimcore/pimcore/pull/14848/files?w=1#diff-c9c26bef59df500e08ac32f8a0882ddbe227bf6661a058d19e9532354848a16cL375 to decide wheter these buttons should appear or not

Reply:
> Please see suggestions from first code scan. Also add migration to mark the bundle as installed on upgrade.

How to handle this for other bundles that got extracted to its own repo? Do we mark them as installed automatically too? 
I know for the web2print it depends on a flag in the settings. They need to add it to the composer json too.

Reply:
We could scan and check if there are any `newsletter` document in the db table, it would be an assumption that it was in use

Reply:
Maybe also check if there are sent mails? Wdyt?

Reply:
not sure, sounds an overkill, would be tied to the fact that email are being sent, there could be differences between env (dev/prod) and where/how this log is stored etc...
Imho the presence of newsletter type is enough to guess that it should warn to install the bundle or the newsletter functionalities won't work propertly